packages/blocks/src/api/serializer.js

```diff
@@ -238,7 +238,8 @@ export function getCommentAttributes( blockType, attributes ) {
 			// Ignore default value.
 			if (
 				'default' in attributeSchema &&
-				attributeSchema.default === value
+				JSON.stringify( attributeSchema.default ) ===
```

Comment:
It would be useful to add test coverage as it might be a bigger issue when dealing with non-empty objects and arrays.

Reply:
The test coverage is there, the `test/integration/full-content` suite does almost 300 tests on various block that are all about parsing and serialization.

Good point about the `{ a, b } === { b, a }` issue. Instead of `JSON.stringify`, we could probably use something like `fast-deep-equal` that sorts object keys before comparing them.

In practice, however, default attribute values tend to be empty-ish objects, and serialization wants to omit values that haven't been modified at all from the default value. So, pragmatically, the `JSON.stringify` solution is likely to be pretty good.

Reply:
> In practice, however, default attribute values tend to be empty-ish objects, and serialization wants to omit values that haven't been modified at all from the default value. So, pragmatically, the `JSON.stringify` solution is likely to be pretty good.

We have an example in the core block (it also applies to its variations, but we override the default value in that case):

https://github.com/WordPress/gutenberg/blob/1bb701445b5ccd45905c0fd1b4d2ffd7395abf4c/packages/block-library/src/query/block.json#L15-L29

However, it's a common issue that plugins encounter, example:

https://github.com/dutchigor/wordpress-block-supports-render-callback-test/blob/3cbef5c6b8cbe95e10979eff5546d10fe5bdb7bb/src/block.json#L21-L28

More details in the following issues:

- https://github.com/WordPress/gutenberg/issues/52600
- https://github.com/WordPress/gutenberg/issues/50229
- https://github.com/WordPress/gutenberg/issues/50027
- https://github.com/WordPress/gutenberg/issues/7342